### ISSUE TITLE
Is a subcomponent flow identifier required to have .path or not?

### ISSUE THREAD

**Issue by stevevestal** [saeaadl/aadlv2.2#13]
For a subcomponent in the middle of a flow, e.g. src -> c1 -> subc -> c2 -> snk, the grammar and some paragraphs allow either subc or subc.path.  For example:

"(N3) The subcomponent flow identifier of a flow implementation must name a subcomponent and optionally a flow specification in the component type of the named subcomponent, or it must name a data component in the form of a data subcomponent, provides data access, or requires data access."

"(4) A flow implementation may refer to subcomponents without identifying a flow specification in the subcomponent. In this case, (a) an unnamed flow specification is assumed to exist from the destination port of the preceding connection to the source port of the succeeding connection if no flow specification is declared for the subcomponent, (b) it represents a separate flow for each of the flow specifications of the subcomponent."

"(3) An end-to-end flow may refer to subcomponents without identifying a flow specification. In this case, the flow specification (a) refers to an unnamed flow specification from the destination port of the preceding connection and the source port of the succeeding connection if the subcomponent has no declared flow specification, or (b) represents a separate end-to-end flow for each of the flow specifications of the subcomponent."

However, there is also the rule

"(N6) In all other cases the subcomponent flow identifier must refer to a flow path or a data component."

and OSATE 2.2.2 issues an error message (even when subc has no flow path declared in its type).

N6 seems to conflict with the other three.  (The "other cases" either refer narrowly to N5 and N4 having to do with source and sink, or more broadly including N3 which which it seems to conflict.)

**Comment by joeseibel** [comment]
Steve, I think you probably stumbled upon the fix for osate/osate2-core#345. Look at the error message that OSATE produces. Does it complain about a connection continuing inside a subcomponent?

**Comment by stevevestal** [comment]
Yes, seeing that error message from OSATE is what triggered this
      escapade.  But while chasing down that problem I looked at the
      standard, and I thought N6 was still somewhat confusing and
      contradictory.  Hence a note to the standards errata but not an
      OSATE issue report.
    
    
    On 8/8/2017 2:35 PM, Joe Seibel wrote:
    
    
      Steve, I think you probably stumbled upon the fix for osate/osate2-core#345. Look at the error message that OSATE produces. Does it complain about a connection continuing inside a subcomponent?


      
      
      
      Steve, I think you probably stumbled upon the fix for osate/osate2-core#345. Look at the
        error message that OSATE produces. Does it complain about a
        connection continuing inside a subcomponent?
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view it on GitHub, or mute the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/saeaadl/aadlv2.2","title":"saeaadl/aadlv2.2","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/saeaadl/aadlv2.2"}},"updates":{"snippets":[{"icon":"PERSON","message":"@joeseibel in #13: Steve, I think you probably stumbled upon the fix for osate/osate2-core#345. Look at the error message that OSATE produces. Does it complain about a connection continuing inside a subcomponent?"}],"action":{"name":"View Issue","url":"https://github.com/saeaadl/aadlv2.2/issues/13#issuecomment-321059126"}}}

**Comment by jjhugues** [comment]
After discussion with Lutz,

(N4)	In case of a flow source implementation the flow identifier of the first subcomponent must refer  to a flow source, a data component, or a subprogram component.
(N5)	In case of a flow sink implementation the flow identifier of the last subcomponent must refer  to a flow sink, a data component, or a subprogram component.
(N6)	In all other cases the subcomponent flow identifier must refer to *a subcomponent,* a flow path, a data component , or subprogram component 

(see other issue #39 )
